### PR TITLE
docs: Add active learning with small text and listener tutorial

### LINKED ISSUE #421: Add active learning with Hugging Face transformers tutorial
Based on the existing active learning tutorial (https://rubrix.readthedocs.io/en/stable/tutorials/05-active_learning.html), we would like to add an example using BaaL and Hugging Face Transformers.

Another option is to extend the current tutorial using Hugging Face and ModAL with skorch. I'd recommend to start with whatever is easier

### LINKED ISSUE #1601: [Listeners] include the active learning tutorial:
See https://github.com/recognai/active-learning-with-rubrix

cc @dcfidalgo

### LINKED ISSUE #1602: Rubrix listeners
A first implementation version that allows clients to react to dataset changes.

These listeners can be the base for other integration like alerts, active learning loops, model monitoring, and other MLOPs actions.

### PR DESCRIPTION
**Pull request by dcfidalgo**

### CODE REVIEW

**Comment by frascuchon** [comment]
@dcfidalgo, please review the PR #1586 before finishing the tutorial. We should change the query definition in this tutorial since record indices are hardly related to active learning batch. 

The `active_learner.query` and `active_learner.update` have a highly coupled, hardly linked to the current batch step. A way to resolve this could be to include the expected batch id as a query param and update it inside the listener action. Let's talk if you want.

**Comment by dcfidalgo** [comment]
See the built docs here: https://rubrix.readthedocs.io/en/docs-add_small_text_with_listener_tutorial/tutorials/active_learning_with_small_text.html

TODO:
 - [x] Add video

File: docs/tutorials/active_learning_with_small_text.ipynb [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+    "rb.log(records, DATASET_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b9142f4-0140-4128-bc90-ae1956190935",
+   "metadata": {},
+   "source": [
+    "Before switching to the Rubrix UI to label the records, we will set up the **active learning loop**. \n",
+    "For this, we will use the [listener decorator](../reference/python/rubrix_listener.rst#rubrix.listeners.listener) from Rubrix. \n",
```
Comment (frascuchon) [inline docs/tutorials/active_learning_with_small_text.ipynb]: This link is broken in the documentation.

See [compiled docs](https://rubrix.readthedocs.io/en/docs-add_small_text_with_listener_tutorial/tutorials/active_learning_with_small_text.html#Rubrix-and-you:-the-perfect-oracle)

**Review by frascuchon — CHANGES_REQUESTED** [review]

File: docs/tutorials/active_learning_with_small_text.ipynb (hunk 1 above)
Comment (dcfidalgo) [reply docs/tutorials/active_learning_with_small_text.ipynb]: done

**Review by frascuchon — APPROVED** [review]

**Pull request merged**